File: output/ifc4x3_dev_73740fe4/entity/IfcAsymmetricIShapeProfileDef.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


public class IfcAsymmetricIShapeProfileDef extends IfcParameterizedProfileDef {

    public IfcPositiveLengthMeasure BottomFlangeWidth;
    public IfcPositiveLengthMeasure OverallDepth;
    public IfcPositiveLengthMeasure WebThickness;
    public IfcPositiveLengthMeasure BottomFlangeThickness;
    public IfcNonNegativeLengthMeasure BottomFlangeFilletRadius; // OPTIONAL
    public IfcPositiveLengthMeasure TopFlangeWidth;
    public IfcPositiveLengthMeasure TopFlangeThickness; // OPTIONAL
    public IfcNonNegativeLengthMeasure TopFlangeFilletRadius; // OPTIONAL
    public IfcNonNegativeLengthMeasure BottomFlangeEdgeRadius; // OPTIONAL
    public IfcPlaneAngleMeasure BottomFlangeSlope; // OPTIONAL
    public IfcNonNegativeLengthMeasure TopFlangeEdgeRadius; // OPTIONAL
    public IfcPlaneAngleMeasure TopFlangeSlope; // OPTIONAL
    // INVERSE attributes:
    // ValidBottomFilletRadius : (NOT(EXISTS(BottomFlangeFilletRadius))) OR
    // (BottomFlangeFilletRadius <= (BottomFlangeWidth - WebThickness)/2.);
    // ValidFlangeThickness : NOT(EXISTS(TopFlangeThickness)) OR ((BottomFlangeThickness + TopFlangeThickness) < OverallDepth);
    // ValidTopFilletRadius : (NOT(EXISTS(TopFlangeFilletRadius))) OR
    // (TopFlangeFilletRadius <= (TopFlangeWidth - WebThickness)/2.);
    // ValidWebThickness : (WebThickness < BottomFlangeWidth) AND (WebThickness < TopFlangeWidth);

    // WHERE constraints:
    // ValidBottomFilletRadius : (NOT(EXISTS(BottomFlangeFilletRadius))) OR
    // (BottomFlangeFilletRadius <= (BottomFlangeWidth - WebThickness)/2.);
    // ValidFlangeThickness : NOT(EXISTS(TopFlangeThickness)) OR ((BottomFlangeThickness + TopFlangeThickness) < OverallDepth);
    // ValidTopFilletRadius : (NOT(EXISTS(TopFlangeFilletRadius))) OR
    // (TopFlangeFilletRadius <= (TopFlangeWidth - WebThickness)/2.);
    // ValidWebThickness : (WebThickness < BottomFlangeWidth) AND (WebThickness < TopFlangeWidth);
}

File: src/main/resources/output/IFC4X3_python/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcAsymmetricIShapeProfileDef/IfcAsymmetricIShapeProfileDef.py
class IfcAsymmetricIShapeProfileDef(IfcParameterizedProfileDef):
    def __init__(self):
        self.BottomFlangeWidth: IfcPositiveLengthMeasure = None
        self.OverallDepth: IfcPositiveLengthMeasure = None
        self.WebThickness: IfcPositiveLengthMeasure = None
        self.BottomFlangeThickness: IfcPositiveLengthMeasure = None
        self.BottomFlangeFilletRadius: IfcNonNegativeLengthMeasure = None
        self.TopFlangeWidth: IfcPositiveLengthMeasure = None
        self.TopFlangeThickness: IfcPositiveLengthMeasure = None
        self.TopFlangeFilletRadius: IfcNonNegativeLengthMeasure = None
        self.BottomFlangeEdgeRadius: IfcNonNegativeLengthMeasure = None
        self.BottomFlangeSlope: IfcPlaneAngleMeasure = None
        self.TopFlangeEdgeRadius: IfcNonNegativeLengthMeasure = None
        self.TopFlangeSlope: IfcPlaneAngleMeasure = None

    # === WHERE CLAUSES ===
    # ValidBottomFilletRadius : (NOT(EXISTS(BottomFlangeFilletRadius))) OR (BottomFlangeFilletRadius <= (BottomFlangeWidth - WebThickness)/2.)
    # ValidFlangeThickness : NOT(EXISTS(TopFlangeThickness)) OR ((BottomFlangeThickness + TopFlangeThickness) < OverallDepth)
    # ValidTopFilletRadius : (NOT(EXISTS(TopFlangeFilletRadius))) OR (TopFlangeFilletRadius <= (TopFlangeWidth - WebThickness)/2.)
    # ValidWebThickness : (WebThickness < BottomFlangeWidth) AND (WebThickness < TopFlangeWidth)
